src/test/java/com/google/cloud/tools/module/DeleteActionTest.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.module;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.GCloudErrorException;
+import com.google.cloud.tools.InvalidFlagException;
+import com.google.cloud.tools.Option;
+import com.google.cloud.tools.ProcessCaller;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Unit tests for {@link DeleteAction}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class DeleteActionTest {
+
+  @Mock
+  private ProcessCaller callerMock;
+
+  @Before
+  public void setUp() throws GCloudErrorException {
+    when(callerMock.getGcloudPath()).thenReturn("here");
+    when(callerMock.call()).thenReturn(true);
+  }
+
+  @Test
+  public void testPrepareCommand() {
+    // Without flags.
+    DeleteAction action = new DeleteAction(
+        ImmutableList.of("module1", "module2"),
+        "v1",
+        ImmutableMap.<Option, String>of());
+
+    Set<String> expected = ImmutableSet.of(action.getProcessCaller().getGcloudPath(), "preview",
+        "app", "modules", "delete", "module1", "module2", "--version", "v1");
+    Set<String> actual = new HashSet<>();
+    actual.addAll(action.getProcessCaller().getCommand());
+    assertEquals(expected, actual);
+
+    // With flags.
+    Map<Option, String> flags = ImmutableMap.of(Option.SERVER, "server.com");
+    action = new DeleteAction(
+        ImmutableList.of("module1", "module2"),
+        "v2",
+        flags);
+    expected = ImmutableSet.of(action.getProcessCaller().getGcloudPath(), "preview", "app",
+        "modules", "delete", "module1", "module2", "--version", "v2", "--server", "server.com");
+    actual = new HashSet<>();
```

Comment:
separate methods (any time you find yourself reusing local variables to point to different things.)